### DISCUSSION TITLE
Gif frames not correctly centered when using -extent - what am I doing wrong?

### DISCUSSION THREAD

**Discussion by lynxmeninx** [ImageMagick/ImageMagick#5342] [Help] [answered]
I'm trying to create some spritesheets out of gifs. They have to have some pretty specific specs - I need all of the frames to be 90x90 pixels, for instance, without changing the appearance of the animation. However, the frames themselves are slightly off-square, so I've been trying to figure out a solution, but I've gotten stuck! I am using ImageMagick 7.1.0 Q16 (windows x64).

Here's the code so far:
```
magick ^
   ( stand*.gif -extent 90x90 -coalesce +append ) ^
   ( walk*.gif -extent 90x90 -coalesce +append ) ^
   -append -filter point -resize 50% -extent 720x720 -gravity northwest spritesheet.png 
```
From two different gifs - one of the character standing, and one of the character walking - the code produces a spritesheet that looks _almost_ perfect. The background is supposed to be transparent, but I've omitted that because it helps to show my issue:
![1](https://user-images.githubusercontent.com/37353809/180097631-87b793cf-6f74-4433-a1b5-86098fadb372.png)
The standing animation seems okay, but there's a problem with both: they are centered on one side of the frame, and the pixels added to 'pad' the edge were only added on one side. From what I can tell, this should be resolveable by using -gravity center, such as in `magick walk*.gif -extent 90x90 -gravity center`. However..
Here's the original gif:
![trotcycle_aloe_right](https://user-images.githubusercontent.com/37353809/180097773-8a429d55-d486-4759-a2fb-6464d1825514.gif)
And here's the gif after using the command above:
![testgravthing](https://user-images.githubusercontent.com/37353809/180097818-93e8de21-7a54-432a-9929-0d011e3a3c51.gif)

I'm not entirely sure what happened here, but it's jittering slightly!
I was able to get the effect I was looking for by using EzGif, which has a gif-to-spritesheet function and also gives you the option to change the size of the 'canvas' for the entire gif; however, it was a bit fiddly with such tiny sprites, and I also have to do it about a hundred more times.  Hence.. imagemagick!

Thank you so much to those who read my last couple of questions, I am in awe of imagemagick and badly want to use it's Sheer and Immense power for this, but I admittedly have zero experience with doing anything like this. I hope I did alright wording all of that!
Thank you again in advance!! ♥

**Comment by snibgo** [comment, marked as answer]
You need `-extent` *after* `-coalesce`, not before.

Before `-coalesce`, frames are probably compressed, so each may be less than the full size of each frame. Extending at that time will give bad results.

EDIT: Similarly, you want `-gravity` to modify the behaviour of `-extent`, so put it before, not after.

**Reply by lynxmeninx** [reply]
You. are. a. life. saver!! That did it!
Thank you so much!
